newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/ExtensionDependenciesEditor.js

```diff
@@ -0,0 +1,168 @@
+//@flow
+import React from 'react';
+import { Trans, t } from '@lingui/macro';
+
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import FlatButton from '../../UI/FlatButton';
+import IconButton from '../../UI/IconButton';
+import Add from '@material-ui/icons/Add';
+import Delete from '@material-ui/icons/Delete';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+
+import Window from '../../Utils/Window';
+import { mapVector } from '../../Utils/MapFor';
+import newNameGenerator from '../../Utils/NewNameGenerator';
+import useForceUpdate from '../../Utils/UseForceUpdate';
+import BackgroundText from '../../UI/BackgroundText';
+
+const checkNameExists = (
+  name: string,
+  deps: gdVectorDependencyMetadata
+): boolean => {
+  for (let i = 0; i < deps.size(); i++)
+    if (deps.at(i).getName() === name) return true;
+  return false;
+};
+
+type Props = {| eventsFunctionsExtension: gdEventsFunctionsExtension |};
+
+export const ExtensionDependenciesEditor = ({
+  eventsFunctionsExtension,
+}: Props) => {
+  const deps = eventsFunctionsExtension.getAllDependencies();
+  const forceUpdate = useForceUpdate();
+
+  const addDependency = () => {
+    eventsFunctionsExtension
+      .addDependency()
+      .setName(
+        newNameGenerator('New Dependency', newName =>
+          checkNameExists(newName, deps)
+        )
+      )
+      .setExportName('my-dependency')
+      .setVersion('1.0.0')
+      .setDependencyType('cordova');
+    forceUpdate();
+  };
+
+  return (
+    <>
+      <TableContainer
+        component={({ children }) => <Paper elevation={4}>{children}</Paper>}
```

Comment:
I don't think you need this props? Just put <Paper> around your <TableContainer>

Reply:
I just followed the examples in the MUI docs:
https://github.com/mui-org/material-ui/blob/48bf11343f04db242b69a0a8ceef3c67ac41de4a/docs/src/pages/components/tables/BasicTable.js#L33